### PR TITLE
Add cache

### PR DESCRIPTION
**Pull request by yourtion**
Read cache `mov` file from `~/Downloads/AerialMovCache/`

### CODE REVIEW

**Comment by yourtion** [comment]
for issues #24

**Comment by JohnCoates** [comment]
I appreciate the intent, but having users download the files themselves, create a directory, etc. is not good for usability. Ideally we want to cache on-demand as the video is being streamed.

**Comment by yourtion** [comment]
@JohnCoates It's the base cache now. My idea is users can download the video themselves because Chinese has bad network.The base case is cache on-demand and user can put the cache file on cache directory manual and I working on it now

**Comment by yourtion** [comment]
@JohnCoates I already finish cache on-demand and user can put the cache file on cache directory manual function. Try it.

**Comment by JohnCoates** [comment]
What your pull request is doing now is this:

#1 if movie has been cached load it from disk
#2 if movie has not been cached, start one stream to play it, start another stream to download it

What would be ideal for #2 is:
if movie has not been cached, start one stream to play it, use the same stream to save it to the cache

**Comment by yourtion** [comment]
Can you merge pull requests first because the user from china can't use it? I will try to make them working in the same stream

File: Readme.md [hunk 1]
```diff
@@ -31,6 +31,16 @@ If Aerial.saver could not be opened, place Aerial.saver in ~/Library/Screen Save
 * **Favorites:** You can choose to only have certain aerials play.
 * **Preview:** Clicking on an aerial in the screen saver options previews that aerial for you.
 
+## Manual Download Files And Add To CacheDirectory
+
+Aerial will cache on-demand and you can add the mov file cache to CacheDirectory.
+
+1. Download the `mov` files form [http://a1.phobos.apple.com/us/r1000/000/Features/atv/AutumnResources/videos/entries.json](http://a1.phobos.apple.com/us/r1000/000/Features/atv/AutumnResources/videos/entries.json) 
+
+2. open System Preferences -> Desktop & Screen Saver -> Screen Saver. Choose Aerial and click on Screen Saver Options and click `OpenCacheDirectory ` and copy the file you download into it.
```
Comment (dimsav) [inline Readme.md]: One space too much?

**Comment by dimsav** [comment]
+1 for adding the self cache option.

I have no idea if that's possible, but for making it user friendlier it would be better to have a nice user interface allowing users to click a "download" button for each video they want to cache.

**Comment by yourtion** [comment]
@dimsav I will try to add it late,it's the most basic cache feature for those person in bad network like me.I add a basic offline mode now.( https://github.com/yourtion/Aerial/releases/download/v1.3/Aerial.saver.zip )

I will try too add the feature to let user start download manual.

**Comment by JohnCoates** [comment]
I went ahead and added single-stream caching. I'll be adding a button to cache everything at once soon.

**Comment by yourtion** [comment]
:+1:

**Pull request closed without merging**